feat: If AppGroup defined in plist apply group dir.

calism2 opened the pull request:
This commit adds functionality to check if an AppGroup is defined in the info.plist. If it is, the group directory is applied, allowing for shared storage between the main app and its extensions.

Closes #702

mrousavy commented on package/cpp/MmkvHostObject.cpp:
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
 
 #ifdef __APPLE__
-  instance = MMKV::mmkvWithID(config.id, mode, encryptionKeyPtr, pathPtr);
+#include <Foundation/Foundation.h>
```
Please move the include up to the other header includes. Don't forget to wrap it in `#ifdef __APPLE__`  again.

mrousavy commented on package/cpp/MmkvHostObject.cpp:
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
-  instance = MMKV::mmkvWithID(config.id, mode, encryptionKeyPtr, pathPtr);
+#include <Foundation/Foundation.h>
+  NSString* appGroup = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"AppGroup"];
+  if (appGroup) {
```
```suggestion
  if (appGroup != nil) {
```

mrousavy commented on package/cpp/MmkvHostObject.cpp:
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
+#include <Foundation/Foundation.h>
+  NSString* appGroup = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"AppGroup"];
+  if (appGroup) {
+    MmkvLogger::log("RNMMKV", "Path ignored, AppGroup defined plist");
```
```suggestion
    MmkvLogger::log("RNMMKV", "Info.plist specifies a custom AppGroup - ignoring `path`...");
```

mrousavy commented on package/cpp/MmkvHostObject.cpp:
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
+    NSString* groupDir =
+        [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:appGroup]
+            .path;
+    const char* groupDirCStr = [groupDir UTF8String];
+    std::string groupDirStr(groupDirCStr);
```
```suggestion
    std::string groupDirStr = groupDir.UTF8String;
```

mrousavy commented on package/cpp/MmkvHostObject.cpp:
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
+    std::string groupDirStr(groupDirCStr);
+
+    instance = MMKV::mmkvWithID(config.id, mode, encryptionKeyPtr, &groupDirStr);
+
```
```suggestion
```

mrousavy reviewed (changes requested):
thanks for the PR! Just a few minor nitpicks.

calism2 commented:
Hi,

Thank you for your fast response. I have implemented the requested changes. Additionally, if the AppGroup capability is not added correctly, the app will throw a null pointer exception, which might be difficult for JavaScript developers to trace. Therefore, I have added a custom runtime error to handle this scenario.

mrousavy commented:
Hey! Thanks for the updates, but I felt like moving `getAppGroup` into `MmkvPlatformContext` made more sense, as it is iOS-specific code and I don't want to cram that into the shared .cpp file.

Let me know what you think: https://github.com/mrousavy/react-native-mmkv/pull/705
